### PR TITLE
Fix: Don't Auto-flatten UIKit Navigation Folders That Only Contain 1 Item

### PR DESCRIPTION
**Pull request by sghoweri**
Updates / reverts the Nav logic in Pattern Lab's UIKit to no longer automatically (slightly) flatten the folder structure when sections only contain a single item. This _might_ resolve the overall problem reported in #1056 by @illepic

### Before (left) vs After (right)
CleanShot 2019-09-17 at 21 48 14@2xCleanShot 2019-09-17 at 21 47 07@2x

### CODE REVIEW

**Comment by illepic** [comment]
@sghoweri I'll give this a shot right now!

File: packages/uikit-workshop/src/scripts/components/pl-nav/pl-nav.js [hunk 1]
```diff
@@ -75,7 +75,7 @@ const SubSubList = props => {
         </Button>
       )}
 
-      {((viewAllItems.length > 0 && nonViewAllItems.length > 1) ||
+      {((viewAllItems.length > 0 && nonViewAllItems.length >= 1) ||
```
Comment (illepic) [inline packages/uikit-workshop/src/scripts/components/pl-nav/pl-nav.js]: `>0` is equivalent to `>= 1`, so you can just write this line: 

```
{((viewAllItems.length && nonViewAllItems.length) ||
```

File: packages/uikit-workshop/src/scripts/components/pl-nav/pl-nav.js [hunk 2]
```diff
@@ -52,7 +52,7 @@ const SubSubList = props => {
               )}
             </a>
 
-            {nonViewAllItems.length > 1 && (
+            {nonViewAllItems.length >= 1 && (
```
Comment (illepic) [inline packages/uikit-workshop/src/scripts/components/pl-nav/pl-nav.js]: Similar to below, just:

```
 {nonViewAllItems.length && (
```

**Comment by illepic** [comment]
Progress!

With a folder structure like so:

image

We've got nav:

image

There are 2 issues, but not major:

1. The listed subtype (derp > derp) is the name of the hidden (underscore) partial, `derp/_derp.twig` instead of the unhidden (non-underscore) template `derp/demo/derps.twig`. Clicking this subtype **does** link to the correct page, however!
1. Clicking the top level pattern results in a non found view-all page:
image

File: packages/uikit-workshop/src/scripts/components/pl-nav/pl-nav.js (hunk 1 above)
Comment (sghoweri) [reply packages/uikit-workshop/src/scripts/components/pl-nav/pl-nav.js]: @illepic lol - you see, this is why I don't do the JavaScripts at 10pm on a work night 😛

File: packages/uikit-workshop/src/scripts/components/pl-nav/pl-nav.js (hunk 2 above)
Comment (sghoweri) [reply packages/uikit-workshop/src/scripts/components/pl-nav/pl-nav.js]: 👍 all set!

**Comment by sghoweri** [comment]
> Progress!
> 
> With a folder structure like so:
> 
> image
> 
> We've got nav:
> 
> image
> 
> There are 2 issues, but not major:
> 
> 1. The listed subtype (derp > derp) is the name of the hidden (underscore) partial, `derp/_derp.twig` instead of the unhidden (non-underscore) template `derp/demo/derps.twig`. Clicking this subtype **does** link to the correct page, however!
> 2. Clicking the top level pattern results in a non found view-all page:
>    image

@illepic what the heck... *scratches head*

Here's what I'm seeing on my end given this folder structure:
CleanShot 2019-09-18 at 07 46 31@2x

### In Pattern Lab PHP (Testing Bolt Codebase)
CleanShot 2019-09-18 at 07 08 51@2x
^ everything works great — the hidden file stays hidden, view all page works, etc

### In Pattern Lab Node (Testing in Twig Edition)
CleanShot 2019-09-18 at 07 45 03@2x
^ I'm strangely _not_ seeing the hidden `derp` file on my end in PL Node... 

Just double checking: there isn't anything funky going on in the YAML or Markdown files that could be rewriting the file, right (frontmatter overrides)? Also, this hidden pattern still incorrectly shows up if you nuke your `dependencyGraph.json` file, yes?

<hr>

Regarding the View All issue, this is definitely a known Pattern Lab Node issue (likely a PL Core issue from what I understand) vs UIKit issue as far as I know. 

FWIW, I think these tickets might be related:
- https://github.com/pattern-lab/patternlab-node/issues/970
- https://github.com/pattern-lab/patternlab-node/issues/890

**Comment by illepic** [comment]
Ah, so here's the kicker: The structure is actually this:

```
derp/
  _derp.twig
  demo/
    derps.twig|md|yml
```

That little "pure" `_derp.twig` _at the root_ of the pattern is key (you have it nested under `demo/` in your example). 

Resulting in:

image

I'm going to remove as many conflating variables as I can here to see what's causing this.

**Comment by coreylafferty** [comment]
@illepic I tested this (using the changes on this PR) and I'm seeing the same thing as @sghoweri.  Looking closely at his screenshot, I think he does have the _derp.twig file in the correct location -- it's not nested under /demo/.  

I'm not sure why you're seeing something different -- but you might try incorporating the changes from this related PR that address an issue when the pattern file name and the folder name match.  

https://github.com/pattern-lab/patternlab-node/pull/1054

@sghoweri it looks like PR 1054 needs the changes you've made here in 1058 in order for the pattern status states to show, so these two PRs should either be merged at once or 1058 should precede the merging of 1054.

**Comment by illepic** [comment]
@coreylafferty great catch! I'll take a look at #1054 .

**Comment by coreylafferty** [comment]
@sghoweri @bmuenzenmeyer Is this PR (and the related changes to core in #1054) good to go?  They both look/work fine for me.  If not, let me know if I can help with any testing or tweaks?

**Comment by sghoweri** [comment]
> @sghoweri @bmuenzenmeyer Is this PR (and the related changes to core in #1054) good to go? They both look/work fine for me. If not, let me know if I can help with any testing or tweaks?

@bmuenzenmeyer Any thoughts on this and #1054? Would love to get these merged down and cut a new UIKit release!

CC @coreylafferty @illepic

**Comment by bmuenzenmeyer** [comment]
@sghoweri FWIW, I've long ago ceded architecture of the front end to you - so if you and multiple other folks assert this is working (hopefully within twig and handlebars) I trust you to merge them

**Comment by sghoweri** [comment]
@bmuenzenmeyer thanks 🙂

Just trying to behave and not get into the habit of merging in my own PRs!

**Comment by bmuenzenmeyer** [comment]
i think i just encountered this myself testing another PR - so will be able to fast-follow test this

**Comment by sghoweri** [comment]
@bmuenzenmeyer I'm resolving the merge conflicts on this then merging this down!

**Comment by bmuenzenmeyer** [comment]
k - would love to try this clean in `dev`

**Comment by sghoweri** [comment]
Tested locally — works great on the PR branch after doing a fresh UIKit build.

**Comment by bmuenzenmeyer** [comment]
post-merge review 👌

**Pull request merged**